tests/Unit/Core/Grid/Column/ColumnCollectionTest.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Unit\Core\Grid\Column;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnInterface;
+use PrestaShop\PrestaShop\Core\Grid\Exception\ColumnNotFoundException;
+
+class ColumnCollectionTest extends TestCase
+{
+    public function testItAddsColumnsToCollection()
+    {
+        $columns = (new ColumnCollection())
+            ->add($this->createColumnMock('first'))
+            ->add($this->createColumnMock('second'))
+            ->add($this->createColumnMock('third'))
+        ;
+
+        $this->assertEquals([
+            'first',
+            'second',
+            'third',
+        ], $this->getColumnPositions($columns));
+
+        return $columns;
+    }
+
+    /**
+     * @depends testItAddsColumnsToCollection
+     */
+    public function testItAddsColumnsBeforeGivenColumn(ColumnCollection $columns)
+    {
+        $columns
+            ->addBefore('first', $this->createColumnMock('before_first'))
+            ->addBefore('second', $this->createColumnMock('before_second'))
+            ->addBefore('third', $this->createColumnMock('before_third'))
+        ;
+
+        $this->assertEquals([
+            'before_first',
+            'first',
+            'before_second',
+            'second',
+            'before_third',
+            'third',
+        ], $this->getColumnPositions($columns));
+
+        return $columns;
+    }
+
+    /**
+     * @depends testItAddsColumnsBeforeGivenColumn
+     */
+    public function testItAddsColumnsAfterGivenColumn(ColumnCollection $columns)
+    {
+        $columns
+            ->addAfter('first', $this->createColumnMock('after_first'))
```

Comment:
Do you have the time to add more tests like `->addBefore()->addAfter()->addBefore()->addAfter()` could be awesome :)

Reply:
sure. :)